File: frontend/lecarden/src/app/core/services/api/basket.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { TokenService } from '../security/token.service';
import { Observable } from 'rxjs';
import { Basket } from 'src/app/shared/models/basket';
import { BasketResult } from 'src/app/shared/models/basketResult';
import { EnvironmentService } from '../helpers/environment.service';
@Injectable({
  providedIn: 'root',
})
export class BasketService {
  private readonly url = `${EnvironmentService.getUrl()}/baskets/baskets`;

  constructor(
    private readonly httpClient: HttpClient,
    private readonly tokenService: TokenService
  ) {}

  getAllPacketsForUser(): Observable<Basket[]> {
    return this.httpClient.get<Basket[]>(
      `${this.url}/user-id/${this.tokenService.getUserId()}`
    );
  }

  updateBaskets(basketResult: BasketResult): Observable<Basket[]> {
    return this.httpClient.post<Basket[]>(this.url, basketResult);
  }

  resetBaskets(packetId: number): Observable<void> {
    return this.httpClient.get<void>(
      `${
        this.url
      }/user-id/${this.tokenService.getUserId()}/packets/${packetId}/reset`
    );
  }
}

File: backend/lecarden/basket-service/src/main/java/com/lecarden/basket/service/BasketService.java
package com.lecarden.basket.service;

import com.lecarden.basket.persistence.to.BasketResult;
import com.lecarden.basket.persistence.to.BasketTO;

import java.util.List;

public interface BasketService {
    BasketTO saveBasket(BasketTO basketTO);

    List<BasketTO> findBasketsByUserId(Long userIdLong);

    List<BasketTO> saveBasket(BasketResult basketResult);

    void resetBaskets(Long userId, Long packetId);
}
